src/System.Linq/tests/MinMaxTests.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class MinMaxTests
+    {
+        [Fact]
+        public void MinInt()
+        {
+            var one = Enumerable.Range(1, 10).ToArray();
+            var minusTen = new[] { -1, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000, 100, 200, 1000 };
+            Assert.Equal(1, one.Min());
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Min());
+        }
+
+        [Fact]
+        public void MaxInt()
+        {
+            var ten = Enumerable.Range(1, 10).ToArray();
+            var minusTen = new[] { -100, -15, -50, -10 };
+            var thousand = new[] { -16, 0, 50, 100, 1000 };
+            Assert.Equal(10, ten.Max());
+            Assert.Equal(-10, minusTen.Max());
+            Assert.Equal(1000, thousand.Max());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Max());
```

Comment:
Similar comments as for MinInt... I'll stop providing the same feedback on the rest of these, and will only note other things if I see any.